### PR TITLE
[expo-store-review] Updated rejection message when review flow fails

### PR DESCRIPTION
**Pull request by mariomurrent-softwaresolutions**
# Why

Because of Issue #16363 

# How

I just added the correct parameters to the promise.reject method according to the implementation

# Test Plan

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by mariomurrent-softwaresolutions** [comment]
@EvanBacon Hope this will be integrated in the next release

File: packages/expo-store-review/android/src/main/java/expo/modules/storereview/StoreReviewModule.kt [hunk 1]
```diff
@@ -53,7 +53,7 @@ class StoreReviewModule(private val mContext: Context) :
           }
         }
       } else {
-        promise.reject(null)
+        promise.reject("ERR_STORE_REVIEW_FAILED", "Android ReviewManager tasks was not successful")
```
Comment (EvanBacon) [inline packages/expo-store-review/android/src/main/java/expo/modules/storereview/StoreReviewModule.kt:56]: Error codes are documented in the [docs](https://docs.expo.dev/versions/latest/sdk/storereview/#error-codes) could you please update the docs to include this.

**Review by EvanBacon — CHANGES_REQUESTED** [review]
see comment

File: packages/expo-store-review/android/src/main/java/expo/modules/storereview/StoreReviewModule.kt (hunk 1 above)
Comment (mariomurrent-softwaresolutions) [reply packages/expo-store-review/android/src/main/java/expo/modules/storereview/StoreReviewModule.kt:56]: Sure, I'll do this tomorrow. Thx for the review

**Comment by mariomurrent-softwaresolutions** [comment]
Updated the DOCs

**Comment by mariomurrent-softwaresolutions** [comment]
Any updates on this one? Will this make it into the new SDK release?

**Comment by bbarthec** [comment]
@mariomurrent-softwaresolutions, I see you're doing some work with this PR, but I also see the enormous number of changes 👀 
Do you mind rebasing with the current `main` branch and cleanup this PR?

**Comment by mariomurrent-softwaresolutions** [comment]
> @mariomurrent-softwaresolutions, I see you're doing some work with this PR, but I also see the enormous number of changes 👀 
> 
> Do you mind rebasing with the current `main` branch and cleanup this PR?

Hi,
Im already on it 😊

**Comment by mariomurrent-softwaresolutions** [comment]
@bbarthec Done

File: packages/expo-store-review/CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- Fixed null pointer exception when store review request failed ([#16365](https://github.com/expo/expo/pull/16365) by [@mariomurrent-softwaresolutions](https://github.com/mariomurrent-softwaresolutions))
```
Comment (bbarthec) [inline packages/expo-store-review/CHANGELOG.md:11]:
```suggestion
- Fixed null pointer exception when store review request failed on Android. ([#16365](https://github.com/expo/expo/pull/16365) by [@mariomurrent-softwaresolutions](https://github.com/mariomurrent-softwaresolutions))
```

File: packages/expo-store-review/android/src/main/java/expo/modules/storereview/StoreReviewModule.kt [hunk 3]
```diff
@@ -53,7 +53,7 @@ class StoreReviewModule(private val mContext: Context) :
           }
         }
       } else {
-        promise.reject(null)
+         promise.reject("ERR_STORE_REVIEW_FAILED", "Android ReviewManager task was not successful")
```
Comment (bbarthec) [inline packages/expo-store-review/android/src/main/java/expo/modules/storereview/StoreReviewModule.kt:56]:
```suggestion
        promise.reject("ERR_STORE_REVIEW_FAILED", "Android ReviewManager task was not successful")
```

**Review by bbarthec — CHANGES_REQUESTED** [review]
Looking a lot better now 😁 
Do not change `react-native-lab/react-native` though 👀 
`git submodule update` or similar would help you with aligning git submodule with the correct commit.

**Comment by mariomurrent-softwaresolutions** [comment]
> Looking a lot better now 😁
> Do not change `react-native-lab/react-native` though 👀
> `git submodule update` or similar would help you with aligning git submodule with the correct commit.

Seems like its easier to clone it freshly and open a new PR ... to much pain with this submodule stuff

**Comment by mariomurrent-softwaresolutions** [comment]
> Looking a lot better now 😁
> Do not change `react-native-lab/react-native` though 👀
> `git submodule update` or similar would help you with aligning git submodule with the correct commit.

Finally I think I reverted the submodule changes

**Comment by mariomurrent-softwaresolutions** [comment]
> @mariomurrent-softwaresolutions, thank you for taking your time and resolving the submodule problem 👏
> Before I can approve this PR I need you to address my suggestions above ☝️

Ah damn, I did not get a notice about the suggestions. Sorry I have totally overseen them

**Review by bbarthec — APPROVED** [review]
Looking good now 😁 
Let us wait for CI and we're ready to merge 😉

**Comment by mariomurrent-softwaresolutions** [comment]
Awesome thx, lets hope the CI will be green as well

**Pull request merged**